chaos/tests/reboot_test.go

```diff
@@ -0,0 +1,95 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/coreos/ktestutil/chaos/cluster"
+	"github.com/coreos/ktestutil/utils"
+
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestClusterMaxDisruption(t *testing.T) {
+	cluster, err := cluster.New(client,
+		cluster.WithMaxDisruption(2),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	doneCh := make(chan struct{})
+	go func() {
+		defer func() { doneCh <- struct{}{} }()
+		if err := cluster.RebootMasters(2 * time.Minute); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	nodes := cluster.Masters
+	hosts := []string{}
+	for i := range nodes {
+		host := utils.ExternalIP(nodes[i])
+		if host == "" {
+			continue
+		}
+		hosts = append(hosts, host)
+	}
+	ssh := utils.MustNewSSHClient(&utils.SSHConfig{Timeout: 10 * time.Second})
+	if err := wait.PollInfinite(10*time.Second, func() (bool, error) {
+		select {
+		case <-doneCh:
+			return true, nil
+		default:
+		}
+
+		rebooting := 0
+		for _, host := range hosts {
+			glog.Infof("running cmd on host: %s", host)
+			_, _, err := ssh.ExecWithTimeout(host, "sudo systemctl is-active kubelet", 10*time.Second)
+			if err != nil {
+				rebooting++
+			}
+		}
+		if rebooting > 1 {
+			return false, fmt.Errorf("more than 1 node rebooting")
+		}
+
+		return false, nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+	close(doneCh)
+}
+
+func TestClusterRebootAll(t *testing.T) {
+	cluster, err := cluster.New(client)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := cluster.RebootAll(2 * time.Minute); err != nil {
```

Comment:
It is not good enough.

When testing func X, we should not assume what func X returns is correct. Relying on the correctness of its return value is wrong.